### PR TITLE
Refactor vsearch

### PR DESCRIPTION
**Pull request by MXueguang**

### CODE REVIEW

File: pyserini/search/nmslib/__init__.py [hunk 1]
```diff
@@ -14,6 +14,6 @@
 # limitations under the License.
 #
 
-from ._vsearcher import SearchResult, SimpleVectorSearcher
+from ._searcher import SearchResult, NmslibSearcher
 
-__all__ = ['SearchResult', 'SimpleVectorSearcher']
+__all__ = ['SearchResult', 'NmslibSearcher']
```
Comment (lintool) [inline pyserini/search/nmslib/__init__.py:19]: `NmslibSearcher` is correct, but looks _ugly_!

File: pyserini/vsearch/index.py [hunk 2]
```diff
@@ -1,101 +0,0 @@
-#
```
Comment (lintool) [inline pyserini/vsearch/index.py:1]: where did this go?

**Comment by lintool** [comment]
Should we add a test case like https://github.com/castorini/pyserini/blob/master/integrations/dense/test_encode.py ?

We're going to need to use `nmslib` at some later point in time... so let's not accrue technical debt?

File: pyserini/search/nmslib/__init__.py (hunk 1 above)
Comment (MXueguang) [reply pyserini/search/nmslib/__init__.py:19]: hrm... have to match Lucene and Faiss i guess

File: pyserini/vsearch/index.py (hunk 2 above)
Comment (MXueguang) [reply pyserini/vsearch/index.py:1]: In pyserini index nmslib already https://github.com/castorini/pyserini/blob/master/pyserini/index/nmslib.py

**Comment by MXueguang** [comment]
> Should we add a test case like https://github.com/castorini/pyserini/blob/master/integrations/dense/test_encode.py ?
> 
> We're going to need to use `nmslib` at some later point in time... so let's not accrue technical debt?

I'll create an issue to track and deal with the test in seperate PR. it will take some time to add a proper integration tests

**Review by lintool — APPROVED** [review]

**Pull request merged**